arcgis-ios-sdk-samples/Geometry/Nearest vertex/NearestVertexViewController.swift

```diff
@@ -90,31 +101,46 @@ class NearestVertexViewController: UIViewController {
+                let nearestCoordinateResult = AGSGeometryEngine.nearestCoordinate(in: polygonGraphic.geometry!, to: normalizedPoint) {
                 // Set the geometry for the tapped point, nearest coordinate point and nearest vertex point.
-                tappedLocationGraphic.geometry = mapPoint
+                tappedLocationGraphic.geometry = normalizedPoint
                 nearestVertexGraphic.geometry = nearestVertexResult.point
                 nearestCoordinateGraphic.geometry = nearestCoordinateResult.point
                 // Get the distance to the nearest vertex in the polygon.
-                let distanceVertex = Measurement(value: nearestVertexResult.distance, unit: UnitLength.meters)
+                let distanceVertex = Measurement(
+                    value: nearestVertexResult.distance,
+                    unit: UnitLength.meters
+                )
                 // Get the distance to the nearest coordinate in the polygon.
-                let distanceCoordinate = Measurement(value: nearestCoordinateResult.distance, unit: UnitLength.meters)
+                let distanceCoordinate = Measurement(
+                    value: nearestCoordinateResult.distance,
+                    unit: UnitLength.meters
+                )
                 // Display the results on a callout of the tapped point.
                 mapView.callout.title = "Proximity result"
-                mapView.callout.detail = String(format: "Vertex dist: %@; Point dist: %@", distanceFormatter.string(from: distanceVertex), distanceFormatter.string(from: distanceCoordinate))
+                mapView.callout.detail = String(
+                    format: "Vertex dist: %@; Point dist: %@",
+                    distanceFormatter.string(from: distanceVertex),
+                    distanceFormatter.string(from: distanceCoordinate)
+                )
                 mapView.callout.isAccessoryButtonHidden = true
-                mapView.callout.show(at: mapPoint, screenOffset: .zero, rotateOffsetWithMap: false, animated: true)
+                mapView.callout.show(
+                    at: normalizedPoint,
+                    screenOffset: .zero,
+                    rotateOffsetWithMap: false,
+                    animated: true
+                )
             } else {
-                // Display the error as an alert if there is a problem with nearestVertex and nearestCoordinate operation.
-                let alertController = UIAlertController(title: nil, message: "Geometry Engine Failed!", preferredStyle: .alert)
-                alertController.addAction(UIAlertAction(title: "OK", style: .default))
-                present(alertController, animated: true)
+                // Display the error as an alert if there is a problem with `AGSGeometryEngine` operations.
+                presentAlert(title: "Error", message: "Geometry Engine Failed!")
```

Comment:
This message could almost certainly be improved. It isn't that the geometry engine failed; it is that it was given bad input. That shouldn't ever happen, as far as I can tell, but the failure condition(s) are not documented. At the very least, it would be good to say what failed: "Could not normalize tapped point."

Reply:
In this case, I'll just totally drop the error message condition.